apps/sel4test-tests/src/tests/scheduler.c

```diff
@@ -1695,3 +1695,72 @@ static int test_changing_affinity(struct env *env)
 }
 DEFINE_TEST(SCHED0022, "test changing a helper threads core", test_changing_affinity,
             (config_set(CONFIG_KERNEL_MCS) &&(CONFIG_MAX_NUM_NODES > 1)));
+
+
+/* busy wait for a number of iterations */
+static void delay_loop(unsigned iterations) {
+    for (int i = 0; i < iterations; i++) {
+        __asm__ __volatile__ ("");
+    }
+}
+
+#define SCHED0023_NUM_ROUNDS 5
+
+/* used in test_round_robin below */
+static void round_robin_thread(int id, volatile unsigned long *counters)
+{
+    delay_loop(id * 100);
+
+    /* assumes counters are initialised already */
+    while (counters[id] < SCHED0023_NUM_ROUNDS) {
+        counters[id]++;
+        /* thread "id" should run after thread "id-1", so we should be at the same
+           count after incrementing if we are scheduled round robin. */
+        if (0 < id) {
+            test_eq(counters[id], counters[id - 1]);
+        }
+        printf("Tick (%d)\n", id);
```

Comment:
Maybe add a explicit note here that the blocking printing is not interfering with the test.

Reply:
I think it will interfere with the test, as it messes up all timing and hence scheduling.